assertj-core/src/main/java/org/assertj/core/api/AbstractByteBufferAssert.java

```diff
@@ -0,0 +1,644 @@
+   *
+   * Example:
+   *
+   * <pre><code class='java'>
+   * ByteBuffer buffer = ByteBuffer.wrap("test".getBytes());
+   *
+   * // this assertion succeeds ...
+   * assertThat(buffer).endsWith("test".getBytes());
+   *
+   * // ... but this one fails as the content of "buffer" does not end with the given expected byte array.
+   * assertThat(buffer).endsWith("test two".getBytes());
+   * </code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} has not been flipped.
+   * @throws AssertionError if the actual {@code ByteBuffer}s content does not end with the given expected byte array.
+   */
+  public SELF endsWith(byte[] expected) {
+    isNotNull();
+    isFlipped();
+
+    byteArrays.assertEndsWith(info, getContent(actual), expected);
+    return myself;
+  }
+
+  /**
+   * Verifies that the content of the actual {@code ByteBuffer} ends with the given expected {@code ByteBuffer}.
+   *
+   * Example:
+   *
+   * <pre><code class='java'>
+   * ByteBuffer actual = ByteBuffer.wrap("test".getBytes());
+   *
+   * // this assertion succeeds ...
+   * ByteBuffer expected = ByteBuffer.wrap("test".getBytes());
+   * assertThat(actual).endsWith(expected);
+   *
+   * // ... but this one fails as the content of "buffer" does not end with the given expected {@code ByteBuffer}.
+   * ByteBuffer expected = ByteBuffer.wrap("test two".getBytes());
+   * assertThat(actual).endsWith(expected);
+   * </code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} has not been flipped.
+   * @throws AssertionError if the actual {@code ByteBuffer}s content does not end with the given expected {@code ByeBuffer}.
+   */
+  public SELF endsWith(ByteBuffer expected) {
+    isNotNull();
+    isFlipped();
+
+    byteArrays.assertEndsWith(info, getContent(actual), getContent(expected));
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code ByteBuffer} is marked.
+   *
+   * Example:
+   *
+   * <pre><code class='java'>
+   * ByteBuffer actual = ByteBuffer.wrap("test".getBytes());
+   *
+   * // this assertion succeeds ...
+   * actual.mark();
+   * assertThat(actual).isMarked();
+   *
+   * // ... but this one fails as "buffer" was never marked.
+   * assertThat(actual).isMarked();
+   * </code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} is not marked.
+   */
+  public SELF isMarked() {
+    isNotNull();
+    try {
+      actual.duplicate().reset();
+    } catch (InvalidMarkException e) {
+      throwAssertionError(shouldBeMarked(actual));
+    }
+    return myself;
+  }
```

Comment:
Unfortunately I had to place the `mark` related assertions on the `ByteBuffer` assertions even though the method is defined on `Buffer`. The reason for this is that the only way I found on how to check if something is marked is by calling `reset()`, which throws the `InvalidMarkException` when the buffer is not marked. But `reset()` modifies the state of the buffer, so I needed a copy of the buffer in order to do so. And while `Buffer` has a `duplicate()` method in more recent versions of Java, in Java 8 (for which AssertJ also has to work if I am not mistaken) the `duplicate()` method was only defined on `ByteBuffer`, and not on `Buffer`.